code/script.js

```diff
@@ -230,96 +323,176 @@ const setSecret = () => {
+    };
+  } 
+   else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
-    }
-  }
-
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
-}
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+    };
+  };
+  generateBoard()
+};
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
+  let personToCheck = personToConfirm
   // store the interaction from the player in a variable.
   // remember the confirm() ?
+  let userConfirmed = window.confirm(`Do you want to guess ${personToConfirm}?`)
   // If the player wants to guess, invoke the checkMyGuess function.
-}
+  if (userConfirmed) {
+    checkMyGuess(personToCheck)
+  }
+  else {
+    alert("Guess cancelled")
+  };
+};
 
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
+  const userHasGuessedCorrectly = (secret.name === personToCheck);
+  displaySecret.src = secret.img
+  secretText.innerHTML = `The mushroom you were identifing was ${secret.name}. Its Swedish name is ${secret.swedishName} and its Latin name is <em>${secret.latinName}</em>. It's easy to find mushrooms in the Swedish forests, but when you identify one you must be sure to look at all its different characteristics, as there are many lookalikes. Don't eat anything if you're not sure!`
+
   // 1. Check if the personToCheck is the same as the secret person's name
   // 2. Set a Message to show in the win or lose section accordingly
+  if (userHasGuessedCorrectly) {
+    winOrLoseText.innerHTML = `Well Done! You checked ${numberOfGuesses} characteristic(s) and correctly identified this mushroom: ${secret.name}!`
+  }
+  else {
+    winOrLoseText.innerHTML = `Oh no! You checked ${numberOfGuesses} characteristic(s) and didn't manage to correctly identify your mushroom today, so don't eat it! It was ${secret.name}. Click below to take another walk!`
+  };
   // 3. Show the win or lose section
+  winOrLoseSection.style.display = 'flex';
+  board.style.display = 'none';
```

Comment:
Do you need both? is it not enough with just displaying the winorlosesection with flex, and that hides the board automatically?